storage/tsdb/tsdb.go

```diff
@@ -26,6 +29,55 @@ import (
 	tsdbLabels "github.com/prometheus/tsdb/labels"
 )
 
+var ErrNotReady = errors.New("TSDB not ready")
+
+type ReadyStorage struct {
+	mtx sync.RWMutex
```

Comment:
The lock and `ErrNotReady` shouldn't really be necessary as no components should try to hit TSDB before it has started, but it's a bit safer and shouldn't cost too much.